opentracing-spring-web-itest/common/src/main/java/io/opentracing/contrib/spring/web/interceptor/itest/common/app/TestController.java

```diff
@@ -0,0 +1,125 @@
+package io.opentracing.contrib.spring.web.interceptor.itest.common.app;
+
+import java.util.concurrent.Callable;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.async.DeferredResult;
+import org.springframework.web.servlet.ModelAndView;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.interceptor.TracingHandlerInterceptor;
+
+/**
+ * @author Pavol Loffay
+ */
+@RestController
+@RequestMapping("/")
+public class TestController {
+
+    public static final String EXCEPTION_MESSAGE = "exceptionMessage";
+    public static final String MAPPED_EXCEPTION_MESSAGE = "mappedExceptionMessage";
+
+    private Tracer tracer;
+
+    @Autowired
+    public TestController(Tracer tracer){
+        this.tracer = tracer;
+    }
+
+    @RequestMapping("/sync")
+    public String sync() {
+        return "sync";
+    }
+
+    @GetMapping(path = "/view")
+    public ModelAndView view() {
+        ModelAndView staticView = new ModelAndView();
+        staticView.setViewName("staticView");
+        return staticView;
+    }
+
+    @RequestMapping("/localSpan")
+    public String localSpan(HttpServletRequest request) {
+        io.opentracing.Tracer.SpanBuilder spanBuilder = tracer.buildSpan("localSpan");
+        Span localSpan = spanBuilder
+                .asChildOf(TracingHandlerInterceptor.serverSpanContext(request))
+                .start();
+        localSpan.finish();
+
+        return "sync";
+    }
+
+    @RequestMapping("/async")
+    public Callable<String> async() {
+        return new Callable<String>() {
+            public String call() throws Exception {
+                Thread.sleep(1000);
```

Comment:
We talked about a similar code in a previous PR, and I have similar concerns here :)